src/ILLink.RoslynAnalyzer/ISymbolExtensions.cs

```diff
@@ -19,5 +21,31 @@ internal static bool HasAttribute (this ISymbol symbol, string attributeName)
 
 			return false;
 		}
+
+		internal static bool TryGetOverriddenMember (this ISymbol? symbol, out ISymbol? overridenMember)
+		{
+			overridenMember = symbol switch {
+				IMethodSymbol method => method.OverriddenMethod,
+				IPropertySymbol property => property.OverriddenProperty,
+				IEventSymbol @event => @event.OverriddenEvent,
+				_ => null,
+			};
+			return overridenMember != null;
+		}
+
+		internal static bool TryGetExplicitOrImplicitInterfaceImplementations (this ISymbol symbol, out ImmutableArray<ISymbol> interfaces)
+		{
+			if (symbol.Kind != SymbolKind.Method && symbol.Kind != SymbolKind.Property && symbol.Kind != SymbolKind.Event)
+				return false;
+
+			var containingType = symbol.ContainingType;
+			var query = from iface in containingType.AllInterfaces
+						from interfaceMember in iface.GetMembers ()
+						let impl = containingType.FindImplementationForInterfaceMember (interfaceMember)
+						where SymbolEqualityComparer.Default.Equals (symbol, impl)
+						select interfaceMember;
+			interfaces = query.ToImmutableArray ();
```

Comment:
We can iterate through the interface members here to save us from Linq :)

Reply:
I would probably treat implicit interface implementations separately since they're tricky to access. Instead of looking at each method, then seeing if it's an interface implementation, why not register a handler for every type, and loop over all implicit interface implementations that are in the same type, and check each one there?

You can probably check explicit implementations along with overrides because they're accessed pretty similarly.